app/src/ui/check-runs/ci-check-run-rerun-dialog.tsx

```diff
@@ -0,0 +1,175 @@
+    await dispatcher.manualRefreshSubscription(
+      repository,
+      prRef,
+      this.state.rerunnable
+    )
+    this.props.onDismissed()
+  }
+
+  private determineRerunnability = async () => {
+    // Get unique set of check suite ids
+    const checkSuiteIds = new Set(
+      this.props.checkRuns.map(cr => cr.checkSuiteId)
+    )
+
+    const checkSuitesPromises = new Array<Promise<IAPICheckSuite | null>>()
+
+    for (const id of checkSuiteIds) {
+      if (id === null) {
+        continue
+      }
+      checkSuitesPromises.push(
+        this.props.dispatcher.fetchCheckSuite(this.props.repository, id)
+      )
+    }
+
+    const rerequestableCheckSuiteIds: number[] = []
+    for (const cs of await Promise.all(checkSuitesPromises)) {
+      if (cs === null) {
+        continue
+      }
+
+      const created = moment(cs.created_at)
+      const monthsSinceCreated = moment().diff(created, 'months')
+      if (
+        cs.rerequestable &&
+        monthsSinceCreated < 1 && // Must be less than a month old
+        cs.status === APICheckStatus.Completed // Must be completed
+      ) {
+        rerequestableCheckSuiteIds.push(cs.id)
+      }
+    }
+
+    const rerunnable = this.props.checkRuns.filter(
+      cr =>
+        cr.checkSuiteId !== null &&
+        rerequestableCheckSuiteIds.includes(cr.checkSuiteId)
+    )
+    const nonRerunnable = this.props.checkRuns.filter(
+      cr =>
+        cr.checkSuiteId === null ||
+        !rerequestableCheckSuiteIds.includes(cr.checkSuiteId)
+    )
+
+    this.setState({ loadingCheckSuites: false, rerunnable, nonRerunnable })
+  }
+
+  private renderRerunnableJobsList = () => {
+    if (this.state.loadingCheckSuites) {
+      return <>Please wait. Determining which checks can be rerun.</>
+    }
+
+    return (
+      <div className="ci-check-run-list check-run-rerun-list">
+        {this.state.rerunnable.length > 0 ? (
+          <CICheckRunList
+            checkRuns={this.state.rerunnable}
+            loadingActionLogs={false}
+            loadingActionWorkflows={false}
+            notExpandable={true}
+          />
+        ) : null}
+      </div>
+    )
+  }
+
+  private renderRerunInfo = () => {
+    if (this.state.loadingCheckSuites) {
+      return null
+    }
+
+    const pluralize = `check${this.state.nonRerunnable.length !== 1 ? 's' : ''}`
+    const verb = this.state.nonRerunnable.length !== 1 ? 'are' : 'is'
+    return (
+      <Row className="non-re-run-info warning-helper-text">
+        <Octicon symbol={OcticonSymbol.alert} />
+
+        {this.state.rerunnable.length === 0
+          ? `There are no checks that can be rerun. `
+          : `There ${verb} ${this.state.nonRerunnable.length} ${pluralize} that cannot be rerun. `}
+
+        {this.state.nonRerunnable.length > 0
+          ? `A check run cannot be rerun if the check is more than one month old,
+          the check has not completed, or the check is not configured to be
+          rerun.`
```

Comment:
Should we use re-run or rerun? (the dialog title, submit button and this message use different spelling).
```suggestion
          ? `There are no checks that can be re-run. `
          : `There ${verb} ${this.state.nonRerunnable.length} ${pluralize} that cannot be re-run. `}

        {this.state.nonRerunnable.length > 0
          ? `A check run cannot be re-run if the check is more than one month old,
          the check has not completed, or the check is not configured to be
          re-run.`
```

Reply:
Good point... and I know from reading stuff on actions... there is debate on should it be.. re-rerun, retry, or rerequest.. :P  But, dotcom uses Re-run so I think it does make sense to follow suite. :)